config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,23 @@
+# Safelist requests from signed in users in the web app
+Rack::Attack.safelist(
+  'mark authenticated request from web app users as safe'
+) do |request|
+  request.env['HTTP_COOKIE'].present? &&
+    request.env['HTTP_AUTHORIZATION'].blank?
+end
+
+# Safelist GET requests
+Rack::Attack.safelist('mark get requests as safe') { |request| request.get? }
+
+# Throttle GraphQL API requests to a configurable value per second that defaults to 20 requests/second
+Rack::Attack.throttle(
+  'limits GraphQL api requests using API token',
+  limit: ENV['GRAPH_API_RATE_LIMIT']&.to_i || 20,
+  period: 1
```

Comment:
Let's set the limit as a more conservative 300 reqs/min?

```suggestion
  limit: ENV['GRAPH_API_RATE_LIMIT']&.to_i || 300,
  period: ENV['GRAPH_API_RATE_PERIOD']&.to_i || 60
```